app/person.rb

```diff
@@ -0,0 +1,21 @@
+class Person
+  attr_accessor :age, :name, :parent_permission
+  attr_reader :id
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def can_use_services?(age)
+    of_age(age) || @parent_permission
+  end
+
+  private
+
+  def of_age?(age)
+    age >= 18
+  end
```

Comment:
- [ ] Kindly don't pass age as an argument because age has been stored in the instance variable `@age`. 🤔 and if you want to use **can_user_services** method then you will need to pass age manually 😨 So your both methods should look similar to below ⬇️

```ruby
  def can_use_services?
    of_age? || @parent_permission
  end

  private

  def of_age?
    @age >= 18
  end
```